src/compiler/checker.ts

```diff
@@ -7780,21 +7782,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (thisParameter) {
                 parameters.unshift(thisParameter);
             }
+            context.flags = flags;

```

Comment:
I think I read this code at some point and the fact that this was never restored was always so odd

Reply:
I think the logic went was always "Suppress implicit any only matters for the lop-level signature, so we don't need to bother" - and that's true.... if the quickfix that sets it (it's only set by a quickfix), is never retooled to use `symbolTableToDeclarationStatements` to do multiple statements at once. If it ever was, then there'd be multiple top-level statements in the context, and that assumption'd be wrong.

So I just made sure we were resetting the flag after changing it, so if we ever do that, we won't get bit.